notebooks/tutorials/coarse_grained_polymer.ipynb

```diff
@@ -0,0 +1,487 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Number of polymer chains\n",
+    "M = 100\n",
+    "\n",
+    "# Number of atoms in each chain\n",
+    "N = 10\n",
+    "\n",
+    "# Add each chain to the topology\n",
+    "for m in range(M):\n",
+    "\n",
+    "    # Make the chain\n",
+    "    chain = topology.addChain()\n",
+    "\n",
+    "    # Create the first residue in the chain\n",
+    "    residue = topology.addResidue(name=\"CG-residue\", chain=chain)\n",
+    "\n",
+    "    # In this example each residue is one bead so we add a single atom\n",
+    "    atom1 = topology.addAtom(name=\"CG-bead\", element=cgElement, residue=residue)\n",
+    "\n",
+    "    # Now add the rest of residues in the chain\n",
+    "    for i in range(1, N):\n",
+    "        residue = topology.addResidue(name=\"CG-residue\",chain=chain)\n",
+    "        atom2 = topology.addAtom(name=\"CG-bead\", element=cgElement, residue=residue)\n",
+    "\n",
+    "        # add the bonds in as we go\n",
+    "        topology.addBond(atom1, atom2)\n",
+    "        atom1 = atom2\n",
+    "\n",
+    "# check the topology\n",
+    "print(topology)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the initial positions\n",
+    "\n",
+    "As we have created the topology from scratch we will also need to define the initial positions from scratch. We will arrange our 100 polymer chains in a 10x10 grid in the X,Y plane and each polymer will be in a linear configuration in the Z direction. Furthermore, we set the simulation box to be cubic with length 11nm."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "positions = []\n",
+    "for m in range(M):\n",
+    "    x0 = np.array(((m%10)*1.0, (m//10)*1.0, 0))\n",
+    "    positions.append(x0)\n",
+    "    for i in range(1,N):\n",
+    "        xi = positions[-1] + np.array((0, 0, 0.38))\n",
+    "        positions.append(xi)\n",
+    "\n",
+    "# convert the list into OpenMM Quantity with units of nm\n",
+    "positions = unit.Quantity(positions, unit.nanometer)\n",
+    "assert(len(positions) == topology.getNumAtoms())\n",
+    "\n",
+    "topology.setPeriodicBoxVectors(np.eye(3)*11.0*unit.nanometers)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can check the initial topology by saving it to a PDB file.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# output the initial configuration\n",
+    "with open('initial_config.pdb','w') as f:\n",
+    "    app.PDBFile.writeFile(topology, positions, f)"
```

Comment:
Note to self: I should make `writeFile()` accept a filename as an alternative to a file object.

Reply:
Yes that would be a useful addition, having to pass a file object is rather cumbersome!